akka-http/src/main/scala/akka/http/scaladsl/server/directives/BasicDirectives.scala

```diff
@@ -238,6 +239,18 @@ trait BasicDirectives {
   def extractMaterializer: Directive1[Materializer] = BasicDirectives._extractMaterializer
 
   /**
+   * Extracts the [[akka.actor.ActorSystem]] from the [[akka.http.scaladsl.server.RequestContext]].
+   *
+   * @group basic
+   */
+  def extractActorSystem: Directive1[ActorSystem] = extract { ctx ⇒
+    ctx.materializer match {
```

Comment:
Instead you can `ActorMaterializerHelper.downcast(ctx.materializer)` (this is private API, but that's fine we're "in" Akka here). It has an unified error message